src/main/java/com/onevizion/scmdb/DbManager.java

```diff
@@ -258,4 +255,50 @@ public void generateDdlForAllObjects() {
         ddlGenerator.executeSettingTransformParams();
         ddlGenerator.generateDllsForAllDbObjects();
     }
+
+    /**
+     * The method filters scripts by type and sorts them in the correct execution order.
+     * Scripts will be sorted in ascending order of numbering, scripts in the range 1..99 will be added to the end of the list.
+     * If the type is ROLLBACK, then the scripts will be sorted in descending order of numbering,
+     * scripts in the range 99..1 will be added to the end of the list.
+     *
+     * Example 1 : Input  scripts = [1_script1.sql, 1_script1_rollback.sql, 8861_script2.sql, 8861_script2_rollback.sql]
+     *                    scriptType = COMMIT
+     *             Output List<SqlScript> = [8861_script2.sql, 1_script1.sql]
+     *
+     * Example 2 : Input  scripts = [1_script1.sql, 1_script1_rollback.sql, 8861_script2.sql, 8861_script2_rollback.sql]
+     *                    scriptType = ROLLBACK
+     *             Output List<SqlScript> = [1_script1_rollback.sql, 8861_script2_rollback.sql]
+     *
+     * @param scripts list of scripts to sort
+     * @param scriptType script type
+     * @return list of scripts with the correct execution order
+     */
+    private List<SqlScript> sortScriptsInExecutionOrder(List<SqlScript> scripts, ScriptType scriptType) {
+        scripts = scripts.stream()
+                         .filter(script -> script.getType() == scriptType)
+                         .collect(Collectors.toList());
+
+        List<SqlScript> commonScripts = new ArrayList<>();
+        List<SqlScript> scriptsLessThanHundred = new ArrayList<>();
+        for (SqlScript script : scripts) {
+            if (script.getName().matches(SCRIPT_NUMBERING_IS_MORE_THAN_TWO_DIGITS_REGEX)) {
```

Comment:
I think this will more readable
![image](https://user-images.githubusercontent.com/3282060/123074348-66f18880-d441-11eb-9d08-3f2fc0b43d8a.png)

Reply:
Done.